Docs Update

cmsirbu opened the pull request:
- Fixed README badge to point at correct project for build status.
- Changed release notes index to reduce duplication.
- Update development Dockerfile to add docs dependencies via poetry
- Major rework of the Developer Guide content - consolidated development environment page and tested all of the steps in a clean VM.

itdependsnetworks commented on development/Dockerfile:
```diff
@@ -41,7 +41,7 @@ COPY . /source
-# TODO: Remove this, once there is a proper local doc hosting strategy
-RUN pip install -r docs/requirements.txt
+# Install build dependencies for docs generation
+RUN poetry add $(cat docs/requirements.txt)
```
This seems odd, is this a preferred pattern.

cmsirbu replied:
The `docs/requirements.txt` is needed by the RTD build process. I don't have a better way of avoiding duplication of pinned deps at the moment.

It can take stuff out of `pyproject.toml` according to their docs but I don't think you want the mkdocs dependencies in the main project deps?

jathanism replied:
The other option is just explicitly adding the requirements to `pyproject.toml` long with `docs/requirements.txt`.

bryanculver replied:
> The other option is just explicitly adding the requirements to `pyproject.toml` along with `docs/requirements.txt`. Reminder to also always use explicitly pinned (`==`) versions for docs dependencies!

This is what core does specifically for this exact reason: a narrow place where duplication does weigh out in complexity.

itdependsnetworks commented:
The temp bad solution on mkdoc requirements is not great here either, but I don't see us as any worse. Going to approve and revisit.

jeffkala commented on docs/dev/dev_environment.md:
```diff
@@ -1,10 +1,138 @@
+## Quickstart
 
-The cookie provides the ability to develop and manage the Nautobot server locally (with supporting services being *Dockerized*) or using only Docker containers to manage Nautobot. The main difference between the two environments is the ability to debug and use **pdb** when developing locally. Debugging with **pdb** within the Docker container is more complicated, but can still be accomplished by  ...
+The development environment can be used in 2 ways:
```
```suggestion
The development environment can be used in two ways:
```

jeffkala commented on docs/dev/dev_environment.md:
```diff
@@ -13,7 +141,7 @@ Follow the directions below for the specific development environment that you ch
 ## Poetry
 
-Poetry is used in lieu of the "virtualenv" commands and is used for either environment. The virtual environment will provide most of the Python packages required to manage the development environment such as **Invoke**, but see the [Local Development Environment](#local-development-environment) section to see how to install Nautobot if you're going to be developing locally. To get started, run th ...
+Poetry is used in lieu of the "virtualenv" commands and is leveraged in both environments. The virtual environment will provide most of the Python packages required to manage the development environment such as **Invoke**, but see the [Local Development Environment](#local-poetry-development-environment) section to see how to install Nautobot if you're going to be developing locally. To get start ...
```
`provides most of the Python packages`? what doesn't it provide?

cmsirbu replied:
Took me a bit to figure it out - it's Nautobot itself. It seems a bit funny the first time, but if you follow the link you'll see that for the non-dockerized setup you have to add nautobot as an extra to the project - and that's not included.

cmsirbu replied:
I've refactored the section.

jeffkala commented on docs/dev/dev_environment.md:
```diff
@@ -26,13 +154,19 @@ The second command puts your shell session into the virtual environment, so all
+
+### Invoke - Building the Docker Image
+
+The first thing you need to do is build the necessary Docker image for Nautobot that installs the specific `nautobot_ver`. The image is used for Nautobot and the RQ worker service used by Docker Compose.
```
```suggestion
The first thing you need to do is build the necessary Docker image for Nautobot that installs the specific `nautobot_ver`. The image is used for Nautobot and the Celery worker service used by Docker Compose.
```

cmsirbu commented:
> The temp bad solution on mkdoc requirements is not great here either, but I don't see us as any worse. Going to approve and revisit.

Based on various discussions and what nautobot-core is doing, copied docs dependencies under `pyproject.toml` under `dev-dependencies`. Tested the Docker container environment and it works like a charm. Also added a section to the docs about updating docs deps versions.

Extra bonus: streamlined the Poetry section and added an invoke task to serve just the docs in local mode.

cmsirbu commented:
Added `test_basic.py` to check `pyproject.toml` and `docs/requirements.txt` are in sync - thanks to @itdependsnetworks !
